extism.go

```diff
@@ -115,6 +115,17 @@ func NewFunction(name string, inputs []ValType, outputs []ValType, f unsafe.Poin
 	return function
 }
 
+func (f *Function) SetNamespace(s string) {
+	cstr := C.CString(s)
+	defer C.free(unsafe.Pointer(cstr))
+	C.extism_function_set_namespace(f.pointer, cstr)
+}
+
+func (f Function) WithNamespace(s string) Function {
```

Comment:
does the method receiver here need to be a pointer type too? wonder or do you know if type checker will fail if its called on a non-pointer Function struct?

Reply:
I see, this is so it can be chained on `NewFunction(...).WithNamespace(...)`.. makes sense